models/_sector/nft/trades/chains/base/nft_base_base_trades.sql

```diff
@@ -0,0 +1,53 @@
+{{ config(
+    schema = 'nft_base',
+    alias = 'base_trades',
+    materialized = 'view'
+    )
+}}
+
+
+{% set nft_models = [
+ ref('alienswap_base_base_trades')
+] %}
+
+with base_union as (
+SELECT * FROM  (
+{% for nft_model in nft_models %}
+    SELECT
+        blockchain,
+        project,
+        project_version,
+        cast(date_trunc('day', block_time) as date) as block_date,
+        cast(date_trunc('month', block_time) as date) as block_month,
+        block_time,
+        block_number,
+        tx_hash,
+        project_contract_address,
+        trade_category,                 --buy/sell/swap
+        trade_type,                     --primary/secondary
+        buyer,
+        seller,
+        nft_contract_address,
+        nft_token_id,
+        nft_amount,                -- always 1 for erc721
+        price_raw,
+        currency_contract,
+        platform_fee_amount_raw,
+        royalty_fee_amount_raw,
+        platform_fee_address,   -- optional
+        royalty_fee_address,    -- optional
+        sub_tx_trade_id,
+--        tx_from,              -- not yet available in the base event tables
+--        tx_to,                -- not yet available in the base event tables
+        row_number() over (partition by tx_hash, sub_tx_trade_id order by tx_hash) as duplicates_rank   -- duplicates protection
+    FROM {{ nft_model }}
+    {% if not loop.last %}
+    UNION ALL
+    {% endif %}
+    {% endfor %}
+    )
+where duplicates_rank = 1
+)
+
+-- this will be removed once tx_from and tx_to are available in the base event tables
+{{ add_nft_tx_data('base_union', 'base') }}
```

Comment:
@jeff-dude 
these columns are provided here with this macro.
But you are right, it's better to do it in the platform models.

I'll quickly adapt it

Reply:
I should have expanded more, the Ethereum version is indeed the same, but all other chains follow this structure (provide the tx_from and tx_to columns in the platform model by using the macro). 
So this is the desired setup. (It's on the todo's to do the same setup on ethereum, but CI was not taking it well the last time I tried)

Reply:
gotcha. i'm for sure confusing myself. in the new dex lineage, we're still running that macro at the chain level, like [here](https://github.com/duneanalytics/spellbook/blob/main/models/_sector/dex/trades/arbitrum/dex_arbitrum_base_trades.sql). ideally we're doing the same across sectors. now that i'm thinking about it, it ultimately doesn't really matter right now, as the long term goal (not so long term i guess, happening now) is to remove completely and have available in decoded, which would mean pulling at platform level.

tl;dr
we're good :)